src/overlays/actors/ovl_En_Sw/z_en_sw.c

```diff
@@ -114,93 +121,93 @@ s32 func_80B0BE20(EnSw* this, CollisionPoly* poly) {
     //! @bug: Does not return.
 }
 
-CollisionPoly* func_80B0C020(PlayState* play, Vec3f* arg1, Vec3f* arg2, Vec3f* arg3, s32* arg4) {
-    CollisionPoly* sp3C;
+CollisionPoly* EnSw_GetPoly(PlayState* play, Vec3f* posA, Vec3f* posB, Vec3f* posOut, s32* bgId) {
+    CollisionPoly* poly;
     s32 pad;
 
-    if (!BgCheck_EntityLineTest1(&play->colCtx, arg1, arg2, arg3, &sp3C, true, true, true, false, arg4)) {
+    if (!BgCheck_EntityLineTest1(&play->colCtx, posA, posB, posOut, &poly, true, true, true, false, bgId)) {
         return NULL;
     }
 
-    if (SurfaceType_GetWallFlags(&play->colCtx, sp3C, *arg4) & WALL_FLAG_CRAWLSPACE) {
+    if (SurfaceType_GetWallFlags(&play->colCtx, poly, *bgId) & WALL_FLAG_CRAWLSPACE) {
         return NULL;
     }
 
-    if (SurfaceType_IsIgnoredByProjectiles(&play->colCtx, sp3C, *arg4)) {
+    if (SurfaceType_IsIgnoredByProjectiles(&play->colCtx, poly, *bgId)) {
         return NULL;
     }
 
-    return sp3C;
+    return poly;
 }
 
-s32 func_80B0C0CC(EnSw* this, PlayState* play, s32 arg2) {
+s32 EnSw_GoldClingToWall(EnSw* this, PlayState* play, s32 arg2) {
     CollisionPoly* temp_v0_2;
     CollisionPoly* temp_s1;
     Vec3f sp9C;
-    Vec3f sp90;
-    Vec3f sp84;
-    Vec3f sp78;
+    Vec3f posOut;
+    Vec3f posA;
+    Vec3f posB;
     s32 pad;
-    s32 sp70;
+    s32 bgId;
     s32 sp6C;
     s32 phi_s1;
-    s32 sp64;
+    s32 ret;
 
-    sp64 = 0;
+    ret = 0;
     this->unk_42C = 1;
-    sp84 = sp78 = this->actor.world.pos;
-    sp84.x += this->unk_364.x * 18.0f;
-    sp84.y += this->unk_364.y * 18.0f;
-    sp84.z += this->unk_364.z * 18.0f;
-    sp78.x -= this->unk_364.x * 18.0f;
-    sp78.y -= this->unk_364.y * 18.0f;
-    sp78.z -= this->unk_364.z * 18.0f;
-    temp_s1 = func_80B0C020(play, &sp84, &sp78, &sp90, &sp70);
-
-    if ((temp_s1 != NULL) && (this->unk_360 == 0)) {
-        sp78.x = sp84.x + (this->unk_37C.x * 24);
-        sp78.y = sp84.y + (this->unk_37C.y * 24);
-        sp78.z = sp84.z + (this->unk_37C.z * 24);
-        temp_v0_2 = func_80B0C020(play, &sp84, &sp78, &sp9C, &sp6C);
+    posA = posB = this->actor.world.pos;
+    posA.x += this->wallPolyNormal.x * 18.0f;
```

Comment:
I would bet this function has shared history with `EnBomChu_Move`. It looks very similar in concept and code

This code isn't wall-specific I think. `wallPolyNormal`, the "up" direction for the skulltula, may well be +y (vertical upwards in world space)

Reply:
bump, I think `wallPolyNormal` and "wall" stuff is misdocumented, pretty sure it applies to floors and possibly ceilings too